### PR TITLE
React testing

### PR DESCRIPTION
**Pull request by blakbox23**
In this part of the project we : 
- Added unit tests for the files `operate.js` and `calculate.js` using Jest,
- Added unit tests for all components,
- Used the react-testing-library to test renders on the UI

### CODE REVIEW

File: src/logic/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    total: 100,
+    next: 10,
+    operation: '÷',
+  };
+  expect(calculate(obj, '÷')).toEqual({
+    total: '10',
+    next: null,
+    operation: '÷',
+  });
+});
```
Comment (Emmyn5600) [inline src/logic/calculate.test.js:62]: - [x] Great job here, all tests are passing successfully, however, let us add few tests to cover all the tests required 

- should throw 0 error if trying to make a zero division
- should throw Zero division error
- should give the modules of the two operand if the buttonName is %
- should display the given value as next (number to be dislayed)

File: src/logic/operate.test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+});
+test('divide 12÷2 to equal 6', () => {
+  expect(operate(12, 2, '÷')).toBe('6');
+});
+test('mutiply 12÷0 to return a 0', () => {
+  expect(operate(0, 7, '÷')).toBe('0');
+});
+test('mutplication 12 X 3 to equal 36', () => {
+  expect(operate(12, 3, 'x')).toBe('36');
+});
```
Comment (Emmyn5600) [inline src/logic/operate.test.js:17]: - [x]  Great job here, all tests are passing successfully, however, let us add few tests to cover all the tests required

- Should give the modulus of the two numbers is the operator is %
- Should throw an error if given a wrong number

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED :recycle:

Hi, @danmainah   and @blakbox23 

Great work on implementing the project requirements, however there are some issues to be solved before you get approved but you are almost there:muscle:. 
   
 - The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- All tests are passing successfully :heavy_check_mark:

rr8


Changes required :writing_hand: 

  - Please check the comment under the review.
  
  Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: src/logic/calculate.test.js (hunk 1 above)
Comment (blakbox23) [reply src/logic/calculate.test.js:62]: Hi @Emmyn5600, thanks for the review, the `calculate.js` doesn't have any errors being thrown. We have however added all the other tests requested.

File: src/logic/operate.test.js (hunk 2 above)
Comment (Emmyn5600) [reply src/logic/operate.test.js:17]: Well done

File: src/logic/calculate.test.js (hunk 1 above)
Comment (Emmyn5600) [reply src/logic/calculate.test.js:62]: Well done guys

**Review by Emmyn5600 — APPROVED** [review]
Hi,

Good job so far! 👏🏼 @danmainah  and @blakbox23 

May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- All tests are passing successfully:heavy_check_mark:

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the changes requested by the previous reviewers.

 #### Optional suggestions
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**